barretenberg/cpp/src/barretenberg/vm/avm/trace/trace.cpp

```diff
@@ -3197,23 +3196,39 @@ AvmError AvmTraceBuilder::op_get_contract_instance(
         error = AvmError::CHECK_TAG_ERROR;
     }
 
-    // Read the contract instance
-    ContractInstanceHint instance = execution_hints.contract_instance_hints.at(read_address.val);
-
-    FF member_value;
-    switch (chosen_member) {
-    case ContractInstanceMember::DEPLOYER:
-        member_value = instance.deployer_addr;
-        break;
-    case ContractInstanceMember::CLASS_ID:
-        member_value = instance.contract_class_id;
-        break;
-    case ContractInstanceMember::INIT_HASH:
-        member_value = instance.initialisation_hash;
-        break;
-    default:
-        member_value = 0;
-        break;
+    FF member_value = 0;
+    bool exists = false;
+
+    if (is_ok(error)) {
+        // Read the contract instance
+        ContractInstanceHint instance = execution_hints.contract_instance_hints.at(read_address.val);
+        // nullifier read hint for the contract address
+        NullifierReadTreeHint nullifier_read_hint = instance.membership_hint;
+        // hinted nullifier should match the specified contract addrss
+        exists = nullifier_read_hint.low_leaf_preimage.nullifier == read_address.val;
+        ASSERT(exists);
+        bool is_member = merkle_tree_trace_builder.perform_nullifier_read(clk,
+                                                                          nullifier_read_hint.low_leaf_preimage,
+                                                                          nullifier_read_hint.low_leaf_index,
+                                                                          nullifier_read_hint.low_leaf_sibling_path);
+        // if the contract address nullifier is hinted as "exists", the membership check should agree
+        ASSERT(is_member == exists);
+        exists = instance.exists;
```

Comment:
I know we still have this `instance.exists` as a throwback to past hinting, but we can't we just rely on this line you had above?
`exists = nullifier_read_hint.low_leaf_preimage.nullifier == read_address.val;`

Reply:
Yes, but until we remove that from the hint, i think it's worth us making sure they align? Otherwise we could end up with ugliness....

Reply:
I took your suggestion. We should just remove the exists hints.